TeamD branches to main branch (#20)

DatBunk opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Moving obstacle that follows a configurable path with selectable interpolation modes, bidirectional traversal, and optional orientation toward movement.
  * Sentient Pin enemy that detects the player, performs autonomous multi-stage jumping attacks, handles collisions, and performs a final exit jump.
  * Tutorial scene demonstrating Sentient Pin behavior.

* **Chores**
  * Added and organized Unity asset folders and metadata for new assets and supporting content.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

MarioVquero committed:
StartofPin

DatBunk committed:
Implement Moving Obstacle/Pin Mechanic

MarioVquero committed:
Jumping works

no gravity yet

MarioVquero committed:
Working Sentient Script

Just needs tweaking

T3C-NEO committed:
Edited the pins

MarioVquero committed:
tweaks and changes to pin

changes to pin from codecks assignment

Make pin properly jump off
Make pin hop forward without the player being in it's detection radius
Switch pin to rigid body on collision

MarioVquero committed:
Cleaned up a little

Brainape committed:
stuyff

MarioVquero committed:
Pin Tweaks

finds new jump spot if stuck, only needs to see player once and bug fixes for weird jumps

Brainape committed:
stuff

MarioVquero committed:
fixed bug of pin getting stuck

stuck pin now attempts rotation before jumping if main jump fails

MarioVquero committed:
Merge branch 'Mario/SentientPin' of https://github.com/StudioAspen/Aspen-2025-2026 into Mario/SentientPin

# Conflicts:
#	charonsCorner/Assets/Scenes/SentientPinTutorial.unity

MarioVquero committed:
tweaks to jump duration

realized having to change both causes the jump coroutine to start mid jump causing errors

MarioVquero committed:
buffer between jumps

added buffer between jumps

Brainape committed:
changes

DatBunk committed:
Moving Obstacle rotation

DatBunk committed:
Merge branch 'GameFeel/TeamD' into Ildefonso/MovingObstacles

samsonahh committed:
Moving Obstacle Rotation - Script Update (#19)

DatBunk committed:
CodeRabbit Suggestions done

Destroyed startpoint objects now, changed comment on line 48, and fixed the lerp issue.

samsonahh committed:
CodeRabbit Suggestions done (#21)

MarioVquero committed:
Merge branch 'GameFeel/TeamD' into Mario/SentientPin

MarioVquero committed:
sentient pin branch to main gamefeel branch (#20)

samsonahh commented:
@coderabbitai can you lint all the .cs files against https://github.com/StudioAspen/Aspen-2025-2026/blob/development/Conventions.md to make sure C# conventions are being met

DatBunk committed:
Update MocingObstacle.cs

DatBunk committed:
Update MovingObstacle.cs

DatBunk committed:
Update MovingObstacle.cs

DatBunk committed:
Update MovingObstacle.cs

MarioVquero committed:
PR cleanup

Added end for infinite loop mentioned by coderabbit, fixed naming conventions on sentient pin script

MarioVquero committed:
Further cleanup

forgot runtime and to make a prefab of the pin

MarioVquero committed:
CodeRabbitchange suggestion

forgot I needed startcoroutine

MarioVquero committed:
Merge branch 'development' into GameFeel/TeamD
